detox/src/devices/android/ADB.js

```diff
@@ -9,7 +10,16 @@ const Environment = require('../../utils/environment');
 class ADB {
   constructor() {
     this._cachedApiLevels = new Map();
-    this.adbBin = path.join(Environment.getAndroidSDKPath(), 'platform-tools', 'adb');
+    try{
+      this.adbBin = path.join(Environment.getAndroidSDKPath(), 'platform-tools', 'adb');
+    } catch (err){
+      const adbOnPath =  which.sync('adb', {nothrow: true});
+      if (adbOnPath) {
+        this.adbBin = adbOnPath;
```

Comment:
Why so many dances? Why not just `this.adbBin = which.sync('adb');`?

Reply:
The reason I went through all this was to prefer the original logic and try that first, and if something wasn't found to throw the same error as detox has been throwing (see below).  I can simplify the code in a new commit.

```
function getAndroidSDKPath() {
  let sdkPath = process.env.ANDROID_SDK_ROOT || process.env.ANDROID_HOME;
  if (!sdkPath) {
    throw new Error(`$ANDROID_SDK_ROOT is not defined, set the path to the SDK installation directory into $ANDROID_SDK_ROOT,
    Go to https://developer.android.com/studio/command-line/variables.html for more details`);
  }
  return sdkPath;
}
```